vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -212,11 +215,14 @@ func getRateLimiter(evictionRateLimit float64, evictionRateLimitBurst int) *rate
 
 // getPodsUpdateOrder returns list of pods that should be updated ordered by update priority
 func (u *updater) getPodsUpdateOrder(pods []*apiv1.Pod, vpa *vpa_types.VerticalPodAutoscaler) []*apiv1.Pod {
-	priorityCalculator := priority.NewUpdatePriorityCalculator(vpa.Spec.ResourcePolicy, vpa.Status.Conditions, nil, u.recommendationProcessor)
-	recommendation := vpa.Status.Recommendation
+	priorityCalculator := priority.NewUpdatePriorityCalculator(
```

Comment:
I'm thinking about making `vpa` part of NewUpdatePriorityCalculator. It seems that we create one calculator per vpa object. Does it make sens?
And then we will not need adding `vpa` to `priorityCalculator.AddPod` signature...

Reply:
Makes all the sense. Applied.